x-pack/metricbeat/module/azure/storage/storage.go

```diff
@@ -41,6 +41,8 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	if err != nil {
 		return nil, err
 	}
+	// set default resource type to indicate this is not the generic monitor metricset
+	ms.Client.Config.DefaultResourceType = defaultStorageAccountNamespace
```

Comment:
is there any way we can test this?

Reply:
unless we change the entire structure of the function and create a mock client etc or to having some integration tests in the future.

Reply:
Looking forward to those!